src/cdk/testing/testing.md

```diff
@@ -1 +1,172 @@
+`ComponentHarness`. All `ComponentHarness` subclasses have a static property, `hostSelector`, that
+matches the harness class to instances of the component in the DOM. Beyond that, the API of any
+given harness is specific to its corresponding component; refer to the component's documentation to
+learn how to use a specific harness.
+
+#### `TestbedHarnessEnvironment` and `ProtractorHarnessEnvironment`
+
+These classes correspond to different implementations of the component harness system with bindings
+for specific test environments. Any given test must only import _one_ of these classes. Karma-based
+unit tests should use the `TestbedHarnessEnvironment`, while Protractor-based end-to-end tests
+should use the `ProtractorHarnessEnvironment`. Additional environments require custom bindings; see
+[API for harness environment authors](#api-for-harness-environment-authors) for more information on
+alternate test environments.
+
+These classes are primarily used to create a `HarnessLoader` instance, and in certain cases, to
+create `ComponentHarness` instances directly.
+
+`TestbedHarnessEnvironment` offers the following static methods:
+
+| Method | Description |
+| ------ | ----------- |
+| `loader(fixture: ComponentFixture<unknown>): HarnessLoader` | Gets a `HarnessLoader` instance for the given fixture, rooted at the fixture's root element. Should be used to create harnesses for elements contained inside the fixture |
+| `documentRootLoader(fixture: ComponentFixture<unknown>): HarnessLoader` | Gets a `HarnessLoader` instance for the given fixture, rooted at the HTML document's root element. Can be used to create harnesses for elements that fall outside of the fixture |
+| `harnessForFixture<T extends ComponentHarness>(fixture: ComponentFixture<unknown>, harnessType: ComponentHarnessConstructor<T>): Promise<T>` | Used to create a `ComponentHarness` instance for the fixture's root element directly. This is necessary when bootstrapping the test with the component you plan to load a harness for, because Angular does not set the proper tag name when creating the fixture. |
+
+In most cases, you can create a `HarnessLoader` in the `beforeEach` block using
+`TestbedHarnessEnvironment.loader(fixture)` and then use that `HarnessLoader` to create any
+necessary `ComponentHarness` instances. The other methods cover special cases as shown in this
+example:  
+
+Consider a reusable dialog-button component that opens a dialog on click, containing the following
+components, each with a corresponding harness:
+- `MyDialogButton` (composes the `MyButton` and `MyDialog` with a convenient API)
+- `MyButton` (a simple button component)
+- `MyDialog` (a dialog appended to `document.body` by `MyButtonDialog` upon click)
+
+The following code loads harnesses for each of these components:
+
+```ts
+it('loads harnesses', async () => {
+  const fixture = TestBed.createComponent(MyDialogButton);
```

Comment:
Above we recommend initializing the harnesses in `beforeEach`, but we aren't showing it here.

Reply:
:+1: its always a battle between making these concise and allowing people to follow good practices by default when copy/pasting